### DISCUSSION TITLE
Can GNPY handle hybrid fiber span?

### DISCUSSION THREAD

**Discussion by caffery-chen** [Telecominfraproject/oopt-gnpy#453] [General]
Dear @AndreaDAmico ,

Can GNPY actually handle the following hybrid link?

1. A mixture of different SMF (e.g., G.652, G.654)
2. A mixture of SSMF and DCF (probably not since GN model is based on the assumption of CD unmanaged link)
3. A mixture of SSMF and NZDCF 

Thanks

**Comment by AndreaDAmico** [comment]
Dear @caffery-chen,

To date, GNPy can handle any hybrid link, but, as you pointed out, with some limitations.
In particular, the NLI solver in GNPy is based on an implementation of the incoherent GN model, which adds up the NLI generated in each span without considering the nontrivial NLI accumulation through multiple spans due to the NLI correlation.
As the NLI is not the main contributor to the GSNR degradation, in the dispersion uncompensated hybrid link scenario this GNPy inaccuracy should not be significant (except for very low dispersion and/or very high baud rate).
On the other hand, in some specific cases, especially with dispersion compensated links (even not hybrid),  the incoherent GN model can be significantly inaccurate. 
In future version, we envision the implementation of accumulation coefficients that enable a better accuracy yet preserving the (spectrally and spatially) disaggregated NLI GN model implementation within GNPy. 
If you are interested, in the following a list of our preliminary investigations:
[1] - A. D’Amico, et al., "Quality of Transmission Estimation for Planning of Disaggregated Optical Networks," 2020 International Conference on Optical Network Design and Modeling (ONDM), 2020
[2] - E. London, et al., "Modelling non-linear interference in non-periodic and disaggregated optical network segments," Opt. Continuum 1, 793-803 (2022)
[3] - E. Virgillito, et al., “Spatially Disaggregated Modeling of Self-Channel NLI in Mixed Fibers Optical Transmission”
